packages/actor-http-native/lib/ActorHttpNative.ts

```diff
@@ -39,22 +39,21 @@ export class ActorHttpNative extends ActorHttp {
     } else {
       options.url = action.input;
     }
+
     if (action.init) {
       Object.assign(options, action.init);
-    }
-    if (options.headers) {
-      const headers: any = {};
-      (<Headers> options.headers).forEach((val: any, key: any) => {
-        headers[key] = val;
-      });
-      options.headers = headers;
     } else {
-      options.headers = {};
-    }
-    if (!options.headers['user-agent']) {
-      options.headers['user-agent'] = this.userAgent;
+      options.headers = (<Request> action.input).headers;
     }
 
+    if (options.headers){
```

Comment:
The following code can be simplified.
Right now you do the same append in two locations.
Instead, you can add an if-check to ensure headers is defined.
After that check, you can do the append.